import/src/test/java/com/google/cloud/healthcare/imaging/dicomadapter/CStoreServiceTest.java

```diff
@@ -287,7 +306,80 @@ public void testCStoreService_transcodeToJpeg2k() throws Exception {
         UID.SecondaryCaptureImageStorage,
         "1.2.276.0.7230010.3.1.4.1784944219.230771.1519337370.699151",
         TestUtils.TEST_IMG_FILE,
-        UID.JPEG2000);
+        UID.JPEG2000,
+        null);
+  }
+
+  @Test
+  public void testBackupUploadService_ExceptionOnWriteToBackup_bytesToStowRsNotAvailable() throws Exception {
+    IBackupUploader mockBackupUploader = mock(IBackupUploader.class);
+    DelayCalculator mockDelayCalculator = mock(DelayCalculator.class);
+    MockStowClient spyStowClient = spy(new MockStowClient(
+        false,
+        HttpStatusCodes.STATUS_CODE_OK));
+
+    doThrow(new IBackupUploader.BackupException("errMsg")).when(mockBackupUploader).doWriteBackup(any(InputStream.class), anyString());
+    doNothing().when(spyStowClient).stowRs(any(InputStream.class));
+
+    BackupUploadService backupUploadService = new BackupUploadService(mockBackupUploader, mockDelayCalculator);
+
+    basicCStoreServiceTest(
+        Status.ProcessingFailure,
+        backupUploadService,
+        spyStowClient);
+
+    verify(mockBackupUploader, never()).doReadBackup(anyString());
+    verify(spyStowClient).stowRs(argThat(is -> equalInputStreamAvailableBytes(is, 0)));
+  }
+
+  @Test
+  public void testBackupUploadService_backupWriteAndReadSuccess() throws Exception {
+    IBackupUploader mockBackupUploader = mock(IBackupUploader.class);
+    DelayCalculator mockDelayCalculator = mock(DelayCalculator.class);
+    MockStowClient spyStowClient = spy(new MockStowClient(
+        false,
+        HttpStatusCodes.STATUS_CODE_OK));
+
+    final InputStream INPUT_STREAM = new ByteArrayInputStream(new byte []{1,2,3,4});
+
+    doAnswer(invocation -> {
+      Object[] args = invocation.getArguments();
+      assertThat(equalInputStreamAvailableBytes(((InputStream)args[0]), 4)).isTrue(); //count available IS bytes before callRealMethod
+      invocation.callRealMethod(); // where bytes is readed
+      return null;
+    }).when(spyStowClient).stowRs(any(InputStream.class));
```

Comment:
Can we remove the equalInputStreamAvailableBytes method here? I'm concerned it may lead to flaky tests. I think verifying that stowrs was called and that we got status success should be enough.